recipes/faac/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+import os
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+
+class FaacConan(ConanFile):
+    name = "faac"
+    description = "Freeware Advanced Audio Coder"
+    topics = ("audio", "mp4", "encoder", "aac", "m4a", "faac")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sourceforge.net/projects/faac"
+    license = "LGPL-2.0-only"
+    exports_sources = "patches/*"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_mp4": [True, False],
+        "drm": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_mp4": False,
+        "drm": False
+    }
+
+    _source_subfolder = "source_subfolder"
+    _autotools = None
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler != "Visual Studio"
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("libfaac doesn't support builing with Visual Studio")
+        if self.options.with_mp4:
+            # TODO: as mpv4v2 as a conan package
+            raise ConanInvalidConfiguration("building with mp4v2 is not supported currently")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        with tools.chdir(os.path.join(self.build_folder, self._source_subfolder)):
+            self.run("./bootstrap", win_bash=tools.os_info.is_windows)
```

Comment:
I would suggest enforcing chmod +x just in case, depending the environment, the user can have a permission error. There is an example [here](https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/meson/all/conanfile.py#L41)